applicationset/services/pull_request/gitea.go

```diff
@@ -68,6 +73,26 @@ func (g *GiteaService) List(ctx context.Context) ([]*PullRequest, error) {
 	return list, nil
 }
 
+// containLabels returns true if gotLabels contains expectedLabels
+func giteaContainLabels(expectedLabels []string, gotLabels []*gitea.Label) bool {
+	for _, expected := range expectedLabels {
+		found := false
+		for _, got := range gotLabels {
+			if got.Name == "" {
+				continue
+			}
+			if expected == got.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
This might be inefficient, tho probably doesn't matter. Can you use a map to check quicker?

Reply:
Fixed.